bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/FileInfoReader.java

```diff
@@ -60,14 +61,13 @@ protected IStatus run(IProgressMonitor monitor) {
 	}
 
 	/**
-	 * Waits until request is processed (barrier[0] is non null).
-	 * This is a bit of a hack, as it would be better if the ECFBrowser worked in similar fashion to
-	 * file transfer were a custom job can be supplied.
-	 * TODO: log an issue for ECF.
+	 * Waits until request is processed (barrierReached.get()). This is a bit of a
+	 * hack, as it would be better if the ECFBrowser worked in similar fashion to
+	 * file transfer were a custom job can be supplied. TODO: log an issue for ECF.
 	 */
 	private void waitOnSelf() {
 		schedule();
-		while (barrier[0] == null) {
+		while (!barrierReached.get()) {
```

Comment:
I must confess that I can't get what is the purpose of this loop... if I schedule a job **once** how can I **join** it more than once?

@merks any clue whats the purpose of this?